backend/admission.go

```diff
@@ -0,0 +1,114 @@
+type AdmissionHandler interface {
+	ValidateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	MutateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	ConvertObject(context.Context, *ConversionRequest) (*AdmissionResponse, error)
+}
+
+type ValidateAdmissionFunc func(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+type MutateAdmissionFunc func(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+type ConvertObjectFunc func(context.Context, *ConversionRequest) (*AdmissionResponse, error)
+
+// Operation is the type of resource operation being checked for admission control
+// https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L158
+type AdmissionRequestOperation int32
+
+const (
+	AdmissionRequestCreate AdmissionRequestOperation = 0
+	AdmissionRequestUpdate AdmissionRequestOperation = 1
+	AdmissionRequestDelete AdmissionRequestOperation = 2
+)
+
+// String textual representation of the operation.
+func (o AdmissionRequestOperation) String() string {
+	return pluginv2.AdmissionRequest_Operation(o).String()
+}
+
+// Identify the Object properties
+type GroupVersionKind struct {
+	Group   string `json:"group,omitempty"`
+	Version string `json:"version,omitempty"`
+	Kind    string `json:"kind,omitempty"`
+}
+
+// AdmissionRequest contains information from a kubernetes Admission request and decoded object(s).
+// See: https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L41
+// And: https://github.com/grafana/grafana-app-sdk/blob/main/resource/admission.go#L14
+// NOTE: this does not include a plugin context
+type AdmissionRequest struct {
+	// NOTE: this may not include app or datasource instance settings depending on the request
+	PluginContext PluginContext `json:"pluginContext,omitempty"`
+	// The requested operation
+	Operation AdmissionRequestOperation `json:"operation,omitempty"`
+	// The object kind
+	Kind GroupVersionKind `json:"kind,omitempty"`
+	// Object is the object in the request.  This includes the full metadata envelope.
+	ObjectBytes []byte `json:"object_bytes,omitempty"`
+	// OldObject is the object as it currently exists in storage. This includes the full metadata envelope.
+	OldObjectBytes []byte `json:"old_object_bytes,omitempty"`
+}
+
+// ConversionRequest supports converting an object from on version to another
+type ConversionRequest struct {
+	// NOTE: this may not include app or datasource instance settings depending on the request
+	PluginContext PluginContext `json:"pluginContext,omitempty"`
+	// The object kind
+	Kind GroupVersionKind `json:"kind,omitempty"`
+	// Object is the object in the request.  This includes the full metadata envelope.
+	ObjectBytes []byte `json:"object_bytes,omitempty"`
+	// Target converted version
+	TargetVersion string `json:"target_version,omitempty"`
+}
+
+// See https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L118
```

Comment:
this link only says "AdmissionResponse describes an admission response" which is not that helpful